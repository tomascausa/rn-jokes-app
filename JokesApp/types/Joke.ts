export type Joke = {
    id: string;
    message: string;
    author?: string;
};
