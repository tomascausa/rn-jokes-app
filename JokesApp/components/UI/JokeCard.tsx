import React from 'react';
import {Text, View, StyleSheet, Dimensions} from 'react-native';
import {GlobalStyles} from '../../constants/styles';
import {Joke} from '../../types/Joke';

export type JokeCardProps = {
    data: Joke;
    mode?: string;
};

export type AuthorProps = {
    label: string;
    style: any;
};

const {width} = Dimensions.get('window');

const Author: React.FC<AuthorProps> = ({label, style}) => {
    return <Text style={style}>@{label}</Text>;
};

const JokeCard: React.FC<JokeCardProps> = ({data, mode}) => {
    let authorElement;

    if (data.permalink) {
        authorElement = (
            <Author
                label={data.permalink}
                style={mode === 'small' ? styles.smallAuthor : styles.author}
            />
        );
    }

    return (
        <View
            style={mode === 'small' ? styles.smallContainer : styles.container}>
            <Text style={mode === 'small' ? styles.smallTitle : styles.title}>
                {data.joke}
            </Text>
            {authorElement}
        </View>
    );
};

const styles = StyleSheet.create({
    smallContainer: {
        flex: 1,
        flexDirection: 'column',
        alignItems: 'center',
        marginHorizontal: 10,
        paddingTop: 35,
        paddingHorizontal: 15,
        paddingBottom: 20,
        shadowColor: GlobalStyles.colors.primary,
        shadowOffset: {
            width: 0,
            height: 15,
        },
        shadowOpacity: 0.3,
        shadowRadius: 10,
        elevation: 15,
        borderRadius: 8,
        backgroundColor: GlobalStyles.colors.primary,
    },
    container: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'space-between',
        width: width - 50,
        marginHorizontal: 20,
        paddingTop: 80,
        paddingBottom: 30,
        paddingHorizontal: 20,
        borderRadius: 8,
        backgroundColor: GlobalStyles.colors.primary,
    },
    title: {
        fontFamily: 'Roboto-Medium',
        fontSize: 30,
        fontWeight: '500',
        lineHeight: 42,
        textAlign: 'center',
        color: GlobalStyles.colors.white,
    },
    smallTitle: {
        fontFamily: 'Roboto-Medium',
        fontSize: 15,
        fontWeight: '500',
        lineHeight: 20,
        textAlign: 'center',
        color: GlobalStyles.colors.white,
    },
    smallAuthor: {
        marginTop: 60,
        fontFamily: 'Roboto-Regular',
        fontSize: 15,
        textAlign: 'center',
        color: GlobalStyles.colors.white,
    },
    author: {
        marginTop: 120,
        fontFamily: 'NanumPenOTF',
        fontSize: 28,
        textAlign: 'center',
        color: GlobalStyles.colors.white,
    },
});

export default JokeCard;
