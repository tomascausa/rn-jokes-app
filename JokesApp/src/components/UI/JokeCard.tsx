import React from 'react';
import {
    Text,
    View,
    StyleSheet,
    StyleProp,
    ViewStyle,
    TextStyle,
} from 'react-native';
import {GlobalStyles} from '../../constants/styles';
import {Joke} from '../../types/Joke';

export type JokeCardProps = {
    data: Joke;
    style?: StyleProp<ViewStyle>;
    titleStyle?: StyleProp<TextStyle>;
    authorStyle?: StyleProp<TextStyle>;
};

export type AuthorProps = {
    label: string;
    style: any;
};

const Author: React.FC<AuthorProps> = ({label, style}) => {
    return <Text style={style}>@{label}</Text>;
};

const JokeCard: React.FC<JokeCardProps> = ({
    data,
    style,
    titleStyle,
    authorStyle,
}) => {
    let authorElement;

    if (data.permalink) {
        authorElement = (
            <Author
                label={data.permalink}
                style={[styles.author, authorStyle]}
            />
        );
    }

    return (
        <View style={[styles.container, style]}>
            <Text style={[styles.title, titleStyle]}>{data.joke}</Text>
            {authorElement}
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'space-between',
    },
    title: {
        fontFamily: 'Roboto-Medium',
        textAlign: 'center',
        color: GlobalStyles.colors.white,
    },
    author: {
        fontFamily: 'NanumPenOTF',
        textAlign: 'center',
        color: GlobalStyles.colors.white,
    },
});

export default JokeCard;
