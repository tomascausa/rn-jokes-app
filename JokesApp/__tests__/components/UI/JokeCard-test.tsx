import 'react-native';
import React from 'react';
import JokeCard from '../../../src/components/UI/JokeCard';
import {Joke} from '../../../src/types/Joke';
import renderer from 'react-test-renderer';

describe('<JokeCard />', () => {
    it('renders correctly', () => {
        const testData: Joke = {
            id: '1',
            joke: 'test',
        };
        const json = renderer.create(<JokeCard data={testData} />).toJSON();
        expect(json).toMatchSnapshot();
    });
});
