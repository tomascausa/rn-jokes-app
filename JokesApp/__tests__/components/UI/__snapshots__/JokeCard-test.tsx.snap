// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JokeCard /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "space-between",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Roboto-Medium",
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    test
  </Text>
</View>
`;
